### PR TITLE
ADP-9462 Feature/azure key vault - Add Azure KeyVault to Easify

### LINKED ISSUE #4: Add support for Azure KeyVault

### PR DESCRIPTION
**Pull request by davidcassell**
# Pull Request Template

## Description

Mo's Azure KeyVault change PR'd

New Features

Closing #4 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Dependent Releases

- Dependent Releases ...

## Required Configuration on CI/CD platform

- Sample Changes

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

Will be tested with FX Exporter.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [ ] My changes fullfill the acceptance criteria for the story or bug
- [ ] I made/planned the required configuration changes on CI/CD

### CODE REVIEW

File: src/Easify.Azure.AspNetCore.AppInsights.UnitTests/ServiceCollectionExtensionsTests.cs [hunk 1]
```diff
@@ -38,5 +38,6 @@ public void Should_ApplicationInsight_IsResolvableFromContainer_WhenTheExtension
             sp.GetRequiredService<ITelemetryInitializer>().Should().NotBeNull().And
                 .BeAssignableTo(typeof(ApplicationInfoTelemetryInitializer));
         }
+
```
Comment (LewisPercival) [inline src/Easify.Azure.AspNetCore.AppInsights.UnitTests/ServiceCollectionExtensionsTests.cs]: JL

File: src/Easify.Azure.AspNetCore.UnitTests/Easify.Azure.AspNetCore.UnitTests.csproj [hunk 2]
```diff
@@ -0,0 +1,32 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+    <PropertyGroup>
+        <TargetFramework>netcoreapp3.1</TargetFramework>
```
Comment (LewisPercival) [inline src/Easify.Azure.AspNetCore.UnitTests/Easify.Azure.AspNetCore.UnitTests.csproj:4]: why didnt we do this in net 5?

**Review by james-heppinstall — APPROVED** [review]
Looks good

**Review by richardcurtis-icg — APPROVED** [review]
Other than the comments from Lewis - looks good to me.

File: src/Easify.Azure.AspNetCore.UnitTests/Easify.Azure.AspNetCore.UnitTests.csproj (hunk 2 above)
Comment (davidcassell) [reply src/Easify.Azure.AspNetCore.UnitTests/Easify.Azure.AspNetCore.UnitTests.csproj:4]: Discussed with Mo, leave this as per 3.1.

**Pull request merged**